Unable to build from source using instructions

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

core

### Describe the bug

I'm unable to build Keycloak from source using the [docs to build from source](https://github.com/keycloak/keycloak/blob/1dd7cf018cb1ce3992b1e75d54b1452ead2d4422/docs/building.md):

OS: Manjaro Linux (Arch-based)

```sh
$ java -version
openjdk version "23" 2024-09-17
OpenJDK Runtime Environment (build 23)
OpenJDK 64-Bit Server VM (build 23, mixed mode, sharing)
```

Also tested on macOS 15.1.1 (same results):

```sh
$ java -version
openjdk version "23.0.1" 2024-10-15
OpenJDK Runtime Environment Homebrew (build 23.0.1)
OpenJDK 64-Bit Server VM Homebrew (build 23.0.1, mixed mode, sharing)
```


### Version

`main`, `26.1.0`, `25.0.6`

### Regression

- [ ] The issue is a regression

### Expected behavior

I expect the project to build successfully and run.

### Actual behavior

Build fails with various errors.

### How to Reproduce?



### `main`

Using the [docs to build from source](https://github.com/keycloak/keycloak/blob/1dd7cf018cb1ce3992b1e75d54b1452ead2d4422/docs/building.md):

<details>
<summary><code>./mvnw clean install</code> output :x:</summary>

```sh
# Git sha: 1dd7cf018cb1ce3992b1e75d54b1452ead2d4422
$ git checkout main
$ ./mvnw clean install
...
[ERROR] Failures:
[ERROR]   SoapTest.test2ConfigurationUsed:172
Expected: a string containing "Connection refused"
     but: was "org.apache.http.conn.ConnectTimeoutException: Connect to localhost:8281 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection timed out"
[ERROR]   TrustedHostClientRegistrationPolicyTest.testGoogleCrawlBot:98 unexpected exception type thrown; expected:<org.keycloak.services.clientregistration.policy.ClientRegistrationPolicyException> but was:<java.lang.NoClassDefFoundError>
[ERROR]   TrustedHostClientRegistrationPolicyTest.testLocalhostDomain:68 unexpected exception type thrown; expected:<org.keycloak.services.clientregistration.policy.ClientRegistrationPolicyException> but was:<java.lang.NoClassDefFoundError>
[ERROR]   TrustedHostClientRegistrationPolicyTest.testLocalhostIP:82 unexpected exception type thrown; expected:<org.keycloak.services.clientregistration.policy.ClientRegistrationPolicyException> but was:<java.lang.NoClassDefFoundError>
[ERROR]   TrustedHostClientRegistrationPolicyTest.testLocalhostName:55 unexpected exception type thrown; expected:<org.keycloak.services.clientregistration.policy.ClientRegistrationPolicyException> but was:<java.lang.ExceptionInInitializerError>
[INFO]
[ERROR] Tests run: 249, Failures: 5, Errors: 0, Skipped: 2
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Keycloak 999.0.0-SNAPSHOT:
[INFO]
[INFO] Keycloak BOM Parent ................................ SUCCESS [  0.708 s]
[INFO] Keycloak BOM for server extensions ................. SUCCESS [  0.026 s]
[INFO] Keycloak ........................................... SUCCESS [  0.037 s]
[INFO] Keycloak Common .................................... SUCCESS [  2.548 s]
[INFO] Keycloak Core ...................................... SUCCESS [  2.749 s]
[INFO] Keycloak Server SPI ................................ SUCCESS [  1.183 s]
[INFO] Keycloak Server Private SPI ........................ SUCCESS [  2.200 s]
[INFO] Keycloak Crypto Parent ............................. SUCCESS [  0.016 s]
[INFO] Keycloak Crypto Default ............................ SUCCESS [  3.796 s]
[INFO] Keycloak SAML Core Public API ...................... SUCCESS [  0.401 s]
[INFO] Keycloak SAML Core ................................. SUCCESS [  2.154 s]
[INFO] Keycloak Model Parent .............................. SUCCESS [  0.007 s]
[INFO] Keycloak Database Support .......................... SUCCESS [  0.121 s]
[INFO] Keycloak Database Support - Private classes ........ SUCCESS [  0.626 s]
[INFO] Keycloak Quarkus Parent ............................ SUCCESS [  0.004 s]
[INFO] Keycloak Configuration API ......................... SUCCESS [  0.417 s]
[INFO] Keycloak REST Services ............................. FAILURE [02:20 min]
[INFO] Keycloak Crypto FIPS 140-2 Integration ............. SKIPPED
[INFO] Keycloak Crypto Wildfly Elytron Integration ........ SKIPPED
[INFO] Keycloak JavaScript Parent ......................... SKIPPED
[INFO] Keycloak Admin UI .................................. SKIPPED
[INFO] Keycloak Account UI ................................ SKIPPED
[INFO] Keycloak Themes Vendor ............................. SKIPPED
[INFO] Keycloak Themes .................................... SKIPPED
[INFO] Keycloak Dependencies Parent ....................... SKIPPED
[INFO] Keycloak Dependencies Server Min ................... SKIPPED
[INFO] Keycloak Model JPA ................................. SKIPPED
[INFO] Keycloak Kerberos Federation ....................... SKIPPED
[INFO] Keycloak LDAP UserStoreProvider .................... SKIPPED
[INFO] Keycloak Database Support - REST services .......... SKIPPED
[INFO] Keycloak Model Infinispan .......................... SKIPPED
[INFO] Keycloak SSSD Federation ........................... SKIPPED
[INFO] Keycloak Ipatuura Federation ....................... SKIPPED
[INFO] Keycloak Authz: Parent ............................. SKIPPED
[INFO] Keycloak AuthZ: Policy Provider Parent ............. SKIPPED
[INFO] Keycloak AuthZ: Common Policy Providers ............ SKIPPED
[INFO] Keycloak Dependencies Server All ................... SKIPPED
[INFO] Keycloak Federation ................................ SKIPPED
[INFO] Keycloak Util Embedded LDAP ........................ SKIPPED
[INFO] Keycloak Util Parent ............................... SKIPPED
[INFO] Keycloak Administration UI ......................... SKIPPED
[INFO] Admin UI REST extensions ........................... SKIPPED
[INFO] Keycloak Integration ............................... SKIPPED
[INFO] Keycloak Admin REST Client for tests ............... SKIPPED
[INFO] Keycloak Client Registration API ................... SKIPPED
[INFO] Keycloak Client CLI ................................ SKIPPED
[INFO] Keycloak Admin CLI ................................. SKIPPED
[INFO] Keycloak Client CLI Distribution ................... SKIPPED
[INFO] Keycloak Authz: Client API ......................... SKIPPED
[INFO] Keycloak JavaScript Admin Client ................... SKIPPED
[INFO] Keycloak UI shared components ...................... SKIPPED
[INFO] Keycloak JavaScript Adapter ........................ SKIPPED
[INFO] Keycloak Test Framework Parent ..................... SKIPPED
[INFO] Keycloak Test Framework - BOM ...................... SKIPPED
[INFO] Keycloak Quarkus Server Extension .................. SKIPPED
[INFO] Keycloak Quarkus Server Deployment ................. SKIPPED
[INFO] Keycloak Quarkus Server App ........................ SKIPPED
[INFO] Keycloak Quarkus Server Distribution ............... SKIPPED
[INFO] Keycloak Quarkus Test Parent ....................... SKIPPED
[INFO] Keycloak Quarkus Server JUnit 5 Internal Test Framework SKIPPED
[INFO] Keycloak Test Framework ............................ SKIPPED
[INFO] Keycloak Test Framework - MariaDB support .......... SKIPPED
[INFO] Keycloak Test Framework - SQL Server support ....... SKIPPED
[INFO] Keycloak Test Framework - MySQL support ............ SKIPPED
[INFO] Keycloak Test Framework - Oracle support ........... SKIPPED
[INFO] Keycloak Test Framework - PostgreSQL support ....... SKIPPED
[INFO] Keycloak Test Framework - Email Server extension ... SKIPPED
[INFO] Keycloak Test Framework Examples Parent ............ SKIPPED
[INFO] Keycloak Test Framework Example Providers .......... SKIPPED
[INFO] Keycloak Test Framework ............................ SKIPPED
[INFO] Keycloak Test Framework ............................ SKIPPED
[INFO] Keycloak Test Framework - Remote providers ......... SKIPPED
[INFO] Keycloak Test Framework - Remote testing utils ..... SKIPPED
[INFO] Keycloak Test Framework Example Tests .............. SKIPPED
[INFO] Keycloak Tests Parent .............................. SKIPPED
[INFO] Keycloak Testsuite Utilities ....................... SKIPPED
[INFO] New Keycloak Base Testsuite ........................ SKIPPED
[INFO] Keycloak Quarkus Server Integration tests .......... SKIPPED
[INFO] Keycloak TestSuite ................................. SKIPPED
[INFO] Keycloak Arquillian Integration TestSuite .......... SKIPPED
[INFO] Servers ............................................ SKIPPED
[INFO] Auth Server ........................................ SKIPPED
[INFO] Auth Server Services ............................... SKIPPED
[INFO] Auth Server Services - Testsuite Providers Deployment SKIPPED
[INFO] Auth Server Services - Testsuite Providers ......... SKIPPED
[INFO] Keycloak Adapter SPI ............................... SKIPPED
[INFO] Test Util .......................................... SKIPPED
[INFO] Keycloak SAML Client Adapter Public API ............ SKIPPED
[INFO] Keycloak SAML Client Adapter Core .................. SKIPPED
[INFO] Test apps .......................................... SKIPPED
[INFO] integration-arquillian-test-apps-servlets .......... SKIPPED
[INFO] Keycloak TestSuite Utils ........................... SKIPPED
[INFO] Auth Server - Undertow ............................. SKIPPED
[INFO] Auth Server - Quarkus .............................. SKIPPED
[INFO] App Server ......................................... SKIPPED
[INFO] App Server - SPI ................................... SKIPPED
[INFO] Keycloak Arquillian Integration TestSuite - Cache Server SKIPPED
[INFO] Keycloak Arquillian Integration TestSuite - Cache Server - Infinispan SKIPPED
[INFO] Keycloak Arquillian Integration TestSuite - Cache Server - Legacy SKIPPED
[INFO] Tests .............................................. SKIPPED
[INFO] Base TestSuite ..................................... SKIPPED
[INFO] Other Tests Modules ................................ SKIPPED
[INFO] Tests for logical storage layer .................... SKIPPED
[INFO] Common JBoss/Wildfly Core Classes .................. SKIPPED
[INFO] Keycloak Client Adapter SPI Modules ................ SKIPPED
[INFO] Keycloak WildFly Elytron SAML Adapter .............. SKIPPED
[INFO] Keycloak Wildfly SAML Adapter Subsystem ............ SKIPPED
[INFO] Keycloak SAML Wildfly Integration .................. SKIPPED
[INFO] Keycloak SAML Client Adapter Modules ............... SKIPPED
[INFO] Keycloak Adapters .................................. SKIPPED
[INFO] Keycloak Docs Parent ............................... SKIPPED
[INFO] Keycloak Guides Maven Plugin ....................... SKIPPED
[INFO] Keycloak Guides .................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:38 min
[INFO] Finished at: 2025-01-16T14:10:32-06:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M7:test (default-test) on project keycloak-services: There are test failures.
[ERROR]
[ERROR] Please refer to /home/eric/Documents/github/keycloak/services/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :keycloak-services
```

(no dump files at `keycloak/services/target/surefire-reports`)

</details>


### `26.1.0`

Using the [docs to build from source](https://github.com/keycloak/keycloak/blob/26.1.0/docs/building.md):


<details>
<summary><code>./mvnw clean install</code> output :x:</summary>

```sh
$ git checkout 26.1.0
$ ./mvnw clean install
...
[ERROR] Failures:
[ERROR]   SoapTest.test2ConfigurationUsed:172
Expected: a string containing "Connection refused"
     but: was "org.apache.http.conn.ConnectTimeoutException: Connect to localhost:8281 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection timed out"
[ERROR]   TrustedHostClientRegistrationPolicyTest.testGoogleCrawlBot:98 unexpected exception type thrown; expected:<org.keycloak.services.clientregistration.policy.ClientRegistrationPolicyException> but was:<java.lang.NoClassDefFoundError>
[ERROR]   TrustedHostClientRegistrationPolicyTest.testLocalhostDomain:68 unexpected exception type thrown; expected:<org.keycloak.services.clientregistration.policy.ClientRegistrationPolicyException> but was:<java.lang.NoClassDefFoundError>
[ERROR]   TrustedHostClientRegistrationPolicyTest.testLocalhostIP:82 unexpected exception type thrown; expected:<org.keycloak.services.clientregistration.policy.ClientRegistrationPolicyException> but was:<java.lang.NoClassDefFoundError>
[ERROR]   TrustedHostClientRegistrationPolicyTest.testLocalhostName:55 unexpected exception type thrown; expected:<org.keycloak.services.clientregistration.policy.ClientRegistrationPolicyException> but was:<java.lang.ExceptionInInitializerError>
[INFO]
[ERROR] Tests run: 249, Failures: 5, Errors: 0, Skipped: 2
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Keycloak 26.1.0:
[INFO]
[INFO] Keycloak BOM Parent ................................ SUCCESS [  0.574 s]
[INFO] Keycloak BOM for server extensions ................. SUCCESS [  0.011 s]
[INFO] Keycloak ........................................... SUCCESS [  0.044 s]
[INFO] Keycloak Common .................................... SUCCESS [  2.546 s]
[INFO] Keycloak Core ...................................... SUCCESS [  3.276 s]
[INFO] Keycloak Server SPI ................................ SUCCESS [  1.171 s]
[INFO] Keycloak Server Private SPI ........................ SUCCESS [  2.020 s]
[INFO] Keycloak Crypto Parent ............................. SUCCESS [  0.005 s]
[INFO] Keycloak Crypto Default ............................ SUCCESS [  4.445 s]
[INFO] Keycloak SAML Core Public API ...................... SUCCESS [  0.404 s]
[INFO] Keycloak SAML Core ................................. SUCCESS [  2.049 s]
[INFO] Keycloak Model Parent .............................. SUCCESS [  0.013 s]
[INFO] Keycloak Database Support .......................... SUCCESS [  0.162 s]
[INFO] Keycloak Database Support - Private classes ........ SUCCESS [  0.847 s]
[INFO] Keycloak Quarkus Parent ............................ SUCCESS [  0.007 s]
[INFO] Keycloak Configuration API ......................... SUCCESS [  0.441 s]
[INFO] Keycloak REST Services ............................. FAILURE [02:21 min]
[INFO] Keycloak Crypto FIPS 140-2 Integration ............. SKIPPED
[INFO] Keycloak Crypto Wildfly Elytron Integration ........ SKIPPED
[INFO] Keycloak JavaScript Parent ......................... SKIPPED
[INFO] Keycloak Admin UI .................................. SKIPPED
[INFO] Keycloak Account UI ................................ SKIPPED
[INFO] Keycloak Themes Vendor ............................. SKIPPED
[INFO] Keycloak Themes .................................... SKIPPED
[INFO] Keycloak Dependencies Parent ....................... SKIPPED
[INFO] Keycloak Dependencies Server Min ................... SKIPPED
[INFO] Keycloak Model JPA ................................. SKIPPED
[INFO] Keycloak Kerberos Federation ....................... SKIPPED
[INFO] Keycloak LDAP UserStoreProvider .................... SKIPPED
[INFO] Keycloak Database Support - REST services .......... SKIPPED
[INFO] Keycloak Model Infinispan .......................... SKIPPED
[INFO] Keycloak SSSD Federation ........................... SKIPPED
[INFO] Keycloak Ipatuura Federation ....................... SKIPPED
[INFO] Keycloak Authz: Parent ............................. SKIPPED
[INFO] Keycloak AuthZ: Policy Provider Parent ............. SKIPPED
[INFO] Keycloak AuthZ: Common Policy Providers ............ SKIPPED
[INFO] Keycloak Dependencies Server All ................... SKIPPED
[INFO] Keycloak Federation ................................ SKIPPED
[INFO] Keycloak Util Embedded LDAP ........................ SKIPPED
[INFO] Keycloak Util Parent ............................... SKIPPED
[INFO] Keycloak Administration UI ......................... SKIPPED
[INFO] Admin UI REST extensions ........................... SKIPPED
[INFO] Keycloak Integration ............................... SKIPPED
[INFO] Keycloak Admin REST Client for tests ............... SKIPPED
[INFO] Keycloak Client Registration API ................... SKIPPED
[INFO] Keycloak Client CLI ................................ SKIPPED
[INFO] Keycloak Admin CLI ................................. SKIPPED
[INFO] Keycloak Client CLI Distribution ................... SKIPPED
[INFO] Keycloak Authz: Client API ......................... SKIPPED
[INFO] Keycloak JavaScript Admin Client ................... SKIPPED
[INFO] Keycloak UI shared components ...................... SKIPPED
[INFO] Keycloak JavaScript Adapter ........................ SKIPPED
[INFO] Keycloak Test Framework Parent ..................... SKIPPED
[INFO] Keycloak Test Framework - BOM ...................... SKIPPED
[INFO] Keycloak Quarkus Server Extension .................. SKIPPED
[INFO] Keycloak Quarkus Server Deployment ................. SKIPPED
[INFO] Keycloak Quarkus Server App ........................ SKIPPED
[INFO] Keycloak Quarkus Server Distribution ............... SKIPPED
[INFO] Keycloak Quarkus Test Parent ....................... SKIPPED
[INFO] Keycloak Quarkus Server JUnit 5 Internal Test Framework SKIPPED
[INFO] Keycloak Test Framework ............................ SKIPPED
[INFO] Keycloak Test Framework - MariaDB support .......... SKIPPED
[INFO] Keycloak Test Framework - SQL Server support ....... SKIPPED
[INFO] Keycloak Test Framework - MySQL support ............ SKIPPED
[INFO] Keycloak Test Framework - Oracle support ........... SKIPPED
[INFO] Keycloak Test Framework - PostgreSQL support ....... SKIPPED
[INFO] Keycloak Test Framework - Email Server extension ... SKIPPED
[INFO] Keycloak Test Framework Examples Parent ............ SKIPPED
[INFO] Keycloak Test Framework Example Providers .......... SKIPPED
[INFO] Keycloak Test Framework ............................ SKIPPED
[INFO] Keycloak Test Framework ............................ SKIPPED
[INFO] Keycloak Test Framework - Remote providers ......... SKIPPED
[INFO] Keycloak Test Framework - Remote testing utils ..... SKIPPED
[INFO] Keycloak Test Framework Example Tests .............. SKIPPED
[INFO] Keycloak Tests Parent .............................. SKIPPED
[INFO] Keycloak Testsuite Utilities ....................... SKIPPED
[INFO] New Keycloak Base Testsuite ........................ SKIPPED
[INFO] Keycloak Quarkus Server Integration tests .......... SKIPPED
[INFO] Keycloak TestSuite ................................. SKIPPED
[INFO] Keycloak Arquillian Integration TestSuite .......... SKIPPED
[INFO] Servers ............................................ SKIPPED
[INFO] Auth Server ........................................ SKIPPED
[INFO] Auth Server Services ............................... SKIPPED
[INFO] Auth Server Services - Testsuite Providers Deployment SKIPPED
[INFO] Auth Server Services - Testsuite Providers ......... SKIPPED
[INFO] Keycloak Adapter SPI ............................... SKIPPED
[INFO] Test Util .......................................... SKIPPED
[INFO] Keycloak SAML Client Adapter Public API ............ SKIPPED
[INFO] Keycloak SAML Client Adapter Core .................. SKIPPED
[INFO] Test apps .......................................... SKIPPED
[INFO] integration-arquillian-test-apps-servlets .......... SKIPPED
[INFO] Keycloak TestSuite Utils ........................... SKIPPED
[INFO] Auth Server - Undertow ............................. SKIPPED
[INFO] Auth Server - Quarkus .............................. SKIPPED
[INFO] App Server ......................................... SKIPPED
[INFO] App Server - SPI ................................... SKIPPED
[INFO] Keycloak Arquillian Integration TestSuite - Cache Server SKIPPED
[INFO] Keycloak Arquillian Integration TestSuite - Cache Server - Infinispan SKIPPED
[INFO] Keycloak Arquillian Integration TestSuite - Cache Server - Legacy SKIPPED
[INFO] Tests .............................................. SKIPPED
[INFO] Base TestSuite ..................................... SKIPPED
[INFO] Other Tests Modules ................................ SKIPPED
[INFO] Tests for logical storage layer .................... SKIPPED
[INFO] Common JBoss/Wildfly Core Classes .................. SKIPPED
[INFO] Keycloak Client Adapter SPI Modules ................ SKIPPED
[INFO] Keycloak WildFly Elytron SAML Adapter .............. SKIPPED
[INFO] Keycloak Wildfly SAML Adapter Subsystem ............ SKIPPED
[INFO] Keycloak SAML Wildfly Integration .................. SKIPPED
[INFO] Keycloak SAML Client Adapter Modules ............... SKIPPED
[INFO] Keycloak Adapters .................................. SKIPPED
[INFO] Keycloak Docs Parent ............................... SKIPPED
[INFO] Keycloak Guides Maven Plugin ....................... SKIPPED
[INFO] Keycloak Guides .................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:40 min
[INFO] Finished at: 2025-01-16T14:17:59-06:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M7:test (default-test) on project keycloak-services: There are test failures.
[ERROR]
[ERROR] Please refer to /home/eric/Documents/github/keycloak/services/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :keycloak-services
```

(no dump files at `keycloak/services/target/surefire-reports`)

</details>


<details>
<summary><code>./mvnw -pl quarkus/deployment,quarkus/dist -am -DskipTests clean install</code> output :x:</summary>

```sh
$ git checkout 26.1.0
$ ./mvnw -pl quarkus/deployment,quarkus/dist -am -DskipTests clean install
...
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /home/eric/Documents/github/keycloak/model/infinispan/src/main/java/org/keycloak/marshalling/KeycloakModelSchema.java:[234,40] cannot find symbol
  symbol:   class KeycloakModelSchemaImpl
  location: interface org.keycloak.marshalling.KeycloakModelSchema
[INFO] 1 error
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Keycloak 26.1.0:
[INFO]
[INFO] Keycloak ........................................... SUCCESS [  0.606 s]
[INFO] Keycloak Common .................................... SUCCESS [  1.986 s]
[INFO] Keycloak Core ...................................... SUCCESS [  1.647 s]
[INFO] Keycloak Server SPI ................................ SUCCESS [  0.836 s]
[INFO] Keycloak Server Private SPI ........................ SUCCESS [  1.451 s]
[INFO] Keycloak Crypto Parent ............................. SUCCESS [  0.018 s]
[INFO] Keycloak Crypto Default ............................ SUCCESS [  0.314 s]
[INFO] Keycloak SAML Core Public API ...................... SUCCESS [  0.409 s]
[INFO] Keycloak SAML Core ................................. SUCCESS [  0.935 s]
[INFO] Keycloak Model Parent .............................. SUCCESS [  0.012 s]
[INFO] Keycloak Database Support .......................... SUCCESS [  0.169 s]
[INFO] Keycloak Database Support - Private classes ........ SUCCESS [  0.447 s]
[INFO] Keycloak Quarkus Parent ............................ SUCCESS [  0.007 s]
[INFO] Keycloak Configuration API ......................... SUCCESS [  0.365 s]
[INFO] Keycloak REST Services ............................. SUCCESS [  4.186 s]
[INFO] Keycloak Crypto FIPS 140-2 Integration ............. SUCCESS [  0.387 s]
[INFO] Keycloak JavaScript Parent ......................... SUCCESS [ 23.717 s]
[INFO] Keycloak Admin UI .................................. SUCCESS [  0.290 s]
[INFO] Keycloak Account UI ................................ SUCCESS [  0.156 s]
[INFO] Keycloak Themes Vendor ............................. SUCCESS [  0.305 s]
[INFO] Keycloak Themes .................................... SUCCESS [  1.241 s]
[INFO] Keycloak Model JPA ................................. SUCCESS [  0.995 s]
[INFO] Keycloak Kerberos Federation ....................... SUCCESS [  0.130 s]
[INFO] Keycloak LDAP UserStoreProvider .................... SUCCESS [  0.472 s]
[INFO] Keycloak Database Support - REST services .......... SUCCESS [  0.153 s]
[INFO] Keycloak Model Infinispan .......................... FAILURE [  0.884 s]
[INFO] Keycloak SSSD Federation ........................... SKIPPED
[INFO] Keycloak Ipatuura Federation ....................... SKIPPED
[INFO] Keycloak Authz: Parent ............................. SKIPPED
[INFO] Keycloak AuthZ: Policy Provider Parent ............. SKIPPED
[INFO] Keycloak AuthZ: Common Policy Providers ............ SKIPPED
[INFO] Keycloak Administration UI ......................... SKIPPED
[INFO] Admin UI REST extensions ........................... SKIPPED
[INFO] Keycloak Integration ............................... SKIPPED
[INFO] Keycloak Client CLI ................................ SKIPPED
[INFO] Keycloak Admin CLI ................................. SKIPPED
[INFO] Keycloak Client CLI Distribution ................... SKIPPED
[INFO] Keycloak Quarkus Server Extension .................. SKIPPED
[INFO] Keycloak Quarkus Server Deployment ................. SKIPPED
[INFO] Keycloak Quarkus Server App ........................ SKIPPED
[INFO] Keycloak Quarkus Server Distribution ............... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  42.793 s
[INFO] Finished at: 2025-01-16T14:27:11-06:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project keycloak-model-infinispan: Compilation failure
[ERROR] /home/eric/Documents/github/keycloak/model/infinispan/src/main/java/org/keycloak/marshalling/KeycloakModelSchema.java:[234,40] cannot find symbol
[ERROR]   symbol:   class KeycloakModelSchemaImpl
[ERROR]   location: interface org.keycloak.marshalling.KeycloakModelSchema
[ERROR]
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :keycloak-model-infinispan
```

</details>


via https://github.com/keycloak/keycloak/issues/24320: `./mvnw -f adapters/oidc/js/pom.xml clean install` doesn't work because `adapters/oidc/js/pom.xml` doesn't exist on this version.



### `25.0.6`

Using the [docs to build from source](https://github.com/keycloak/keycloak/blob/25.0.6/docs/building.md):


<details>
<summary><code>./mvnw clean install</code> output :x:</summary>

```sh
$ git checkout 25.0.6
$ ./mvnw clean install
...
[ERROR] Errors:
[ERROR]   CertificateValidatorTest.testCertificatePolicyModeAllNotRequestedAndNotPresent:119->testCertificatePolicyValidation:347 » NoClassDefFound Could not initialize class org.keycloak.authentication.authenticators.x509.CertificateValidator
[ERROR]   CertificateValidatorTest.testCertificatePolicyModeAllNotRequestedAndOnePresent:128->testCertificatePolicyValidation:347 » ExceptionInInitializer
[ERROR]   CertificateValidatorTest.testCertificatePolicyModeAllNotRequestedAndTwoPresent:137->testCertificatePolicyValidation:347 » NoClassDefFound Could not initialize class org.keycloak.authentication.authenticators.x509.CertificateValidator
[ERROR]   CertificateValidatorTest.testCertificatePolicyModeAllOneRequestedAndNotPresent »  Unexpected exception, expected<java.security.GeneralSecurityException> but was<java.lang.NoClassDefFoundError>
[ERROR]   CertificateValidatorTest.testCertificatePolicyModeAllOneRequestedAndOnePresent:155->testCertificatePolicyValidation:347 » NoClassDefFound Could not initialize class org.keycloak.authentication.authenticators.x509.CertificateValidator
[ERROR]   CertificateValidatorTest.testCertificatePolicyModeAllOneRequestedAndOnePresentDifferent »  Unexpected exception, expected<java.security.GeneralSecurityException> but was<java.lang.NoClassDefFoundError>
[ERROR]   CertificateValidatorTest.testCertificatePolicyModeAllOneRequestedAndTwoPresent:173->testCertificatePolicyValidation:347 » NoClassDefFound Could not initialize class org.keycloak.authentication.authenticators.x509.CertificateValidator
[ERROR]   CertificateValidatorTest.testCertificatePolicyModeAllOneRequestedAndTwoPresentDifferent »  Unexpected exception, expected<java.security.GeneralSecurityException> but was<java.lang.NoClassDefFoundError>
[ERROR]   CertificateValidatorTest.testCertificatePolicyModeAllTwoRequestedAndNotPresent »  Unexpected exception, expected<java.security.GeneralSecurityException> but was<java.lang.NoClassDefFoundError>
[ERROR]   CertificateValidatorTest.testCertificatePolicyModeAllTwoRequestedAndOnePresent »  Unexpected exception, expected<java.security.GeneralSecurityException> but was<java.lang.NoClassDefFoundError>
[ERROR]   CertificateValidatorTest.testCertificatePolicyModeAllTwoRequestedAndOnePresentDifferent »  Unexpected exception, expected<java.security.GeneralSecurityException> but was<java.lang.NoClassDefFoundError>
[ERROR]   CertificateValidatorTest.testCertificatePolicyModeAllTwoRequestedAndTwoPresent:218->testCertificatePolicyValidation:347 » NoClassDefFound Could not initialize class org.keycloak.authentication.authenticators.x509.CertificateValidator
[ERROR]   CertificateValidatorTest.testCertificatePolicyModeAnyNotRequestedAndNotPresent:227->testCertificatePolicyValidation:347 » NoClassDefFound Could not initialize class org.keycloak.authentication.authenticators.x509.CertificateValidator
[ERROR]   CertificateValidatorTest.testCertificatePolicyModeAnyNotRequestedAndOnePresent:236->testCertificatePolicyValidation:347 » NoClassDefFound Could not initialize class org.keycloak.authentication.authenticators.x509.CertificateValidator
[ERROR]   CertificateValidatorTest.testCertificatePolicyModeAnyNotRequestedAndTwoPresent:245->testCertificatePolicyValidation:347 » NoClassDefFound Could not initialize class org.keycloak.authentication.authenticators.x509.CertificateValidator
[ERROR]   CertificateValidatorTest.testCertificatePolicyModeAnyOneRequestedAndNotPresent »  Unexpected exception, expected<java.security.GeneralSecurityException> but was<java.lang.NoClassDefFoundError>
[ERROR]   CertificateValidatorTest.testCertificatePolicyModeAnyOneRequestedAndOnePresent:263->testCertificatePolicyValidation:347 » NoClassDefFound Could not initialize class org.keycloak.authentication.authenticators.x509.CertificateValidator
[ERROR]   CertificateValidatorTest.testCertificatePolicyModeAnyOneRequestedAndOnePresentDifferent »  Unexpected exception, expected<java.security.GeneralSecurityException> but was<java.lang.NoClassDefFoundError>
[ERROR]   CertificateValidatorTest.testCertificatePolicyModeAnyOneRequestedAndTwoPresent:281->testCertificatePolicyValidation:347 » NoClassDefFound Could not initialize class org.keycloak.authentication.authenticators.x509.CertificateValidator
[ERROR]   CertificateValidatorTest.testCertificatePolicyModeAnyOneRequestedAndTwoPresentDifferent »  Unexpected exception, expected<java.security.GeneralSecurityException> but was<java.lang.NoClassDefFoundError>
[ERROR]   CertificateValidatorTest.testCertificatePolicyModeAnyTwoRequestedAndNotPresent »  Unexpected exception, expected<java.security.GeneralSecurityException> but was<java.lang.NoClassDefFoundError>
[ERROR]   CertificateValidatorTest.testCertificatePolicyModeAnyTwoRequestedAndOnePresent:308->testCertificatePolicyValidation:347 » NoClassDefFound Could not initialize class org.keycloak.authentication.authenticators.x509.CertificateValidator
[ERROR]   CertificateValidatorTest.testCertificatePolicyModeAnyTwoRequestedAndOnePresentDifferent »  Unexpected exception, expected<java.security.GeneralSecurityException> but was<java.lang.NoClassDefFoundError>
[ERROR]   CertificateValidatorTest.testCertificatePolicyModeAnyTwoRequestedAndTwoPresent:326->testCertificatePolicyValidation:347 » NoClassDefFound Could not initialize class org.keycloak.authentication.authenticators.x509.CertificateValidator
[ERROR]   CertificateValidatorTest.testValidityOfCertificatesHasExpired:103 » NoClassDefFound Could not initialize class org.keycloak.authentication.authenticators.x509.CertificateValidator
[ERROR]   CertificateValidatorTest.testValidityOfCertificatesNotValidYet:75 » NoClassDefFound Could not initialize class org.keycloak.authentication.authenticators.x509.CertificateValidator
[ERROR]   CertificateValidatorTest.testValidityOfCertificatesSuccess:49 » NoClassDefFound Could not initialize class org.keycloak.authentication.authenticators.x509.CertificateValidator
[ERROR]   RedirectUtilsTest.beforeClass:51 » ServiceConfiguration org.keycloak.authentication.AuthenticatorFactory: Provider org.keycloak.authentication.authenticators.browser.UsernamePasswordFormFactory could not be instantiated
[ERROR]   SoapTest.test1ResponseOK:116 » ServiceConfiguration org.keycloak.authentication.AuthenticatorFactory: Provider org.keycloak.authentication.authenticators.browser.UsernamePasswordFormFactory could not be instantiated
[ERROR]   SoapTest.test2ConfigurationUsed:163 » ServiceConfiguration org.keycloak.authentication.ClientAuthenticatorFactory: Provider org.keycloak.authentication.authenticators.client.X509ClientAuthenticator could not be instantiated
[ERROR]   TrustedHostClientRegistrationPolicyTest.beforeClass:46 » ServiceConfiguration org.keycloak.authentication.AuthenticatorFactory: Provider org.keycloak.authentication.authenticators.browser.UsernamePasswordFormFactory could not be instantiated
[ERROR]   DefaultSecurityProfileProverFactoryTest.beforeClass:68 » ServiceConfiguration org.keycloak.authentication.ClientAuthenticatorFactory: Provider org.keycloak.authentication.authenticators.client.X509ClientAuthenticator could not be instantiated
[INFO]
[ERROR] Tests run: 208, Failures: 0, Errors: 32, Skipped: 2
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Keycloak 25.0.6:
[INFO]
[INFO] Keycloak BOM Parent ................................ SUCCESS [  0.496 s]
[INFO] Keycloak BOM for adapters .......................... SUCCESS [  0.005 s]
[INFO] Keycloak BOM for server extensions ................. SUCCESS [  0.004 s]
[INFO] Keycloak BOM utilities for the quickstarts ......... SUCCESS [  0.004 s]
[INFO] Keycloak ........................................... SUCCESS [  3.049 s]
[INFO] Keycloak Common .................................... SUCCESS [  3.099 s]
[INFO] Keycloak Core ...................................... SUCCESS [  3.238 s]
[INFO] Keycloak Server SPI ................................ SUCCESS [  1.399 s]
[INFO] Keycloak Server Private SPI ........................ SUCCESS [  2.200 s]
[INFO] Keycloak Crypto Parent ............................. SUCCESS [  0.011 s]
[INFO] Keycloak Crypto Default ............................ SUCCESS [  3.400 s]
[INFO] Keycloak SAML Core Public API ...................... SUCCESS [  0.370 s]
[INFO] Keycloak SAML Core ................................. SUCCESS [  2.130 s]
[INFO] Keycloak Model Parent .............................. SUCCESS [  0.020 s]
[INFO] Keycloak Database Support .......................... SUCCESS [  0.153 s]
[INFO] Keycloak Database Support - Private classes ........ SUCCESS [  0.627 s]
[INFO] Keycloak REST Services ............................. FAILURE [  6.298 s]
[INFO] Keycloak Crypto FIPS 140-2 Integration ............. SKIPPED
[INFO] Keycloak Crypto Wildfly Elytron Integration ........ SKIPPED
[INFO] Keycloak JavaScript Adapter (JAR) .................. SKIPPED
[INFO] Keycloak JavaScript Parent ......................... SKIPPED
[INFO] Keycloak Admin UI .................................. SKIPPED
[INFO] Keycloak Account UI ................................ SKIPPED
[INFO] Keycloak Themes .................................... SKIPPED
[INFO] Keycloak Dependencies Parent ....................... SKIPPED
[INFO] Keycloak Dependencies Server Min ................... SKIPPED
[INFO] Keycloak Model JPA ................................. SKIPPED
[INFO] Keycloak Kerberos Federation ....................... SKIPPED
[INFO] Keycloak LDAP UserStoreProvider .................... SKIPPED
[INFO] Keycloak Database Support - REST services .......... SKIPPED
[INFO] Keycloak Model Infinispan .......................... SKIPPED
[INFO] Keycloak SSSD Federation ........................... SKIPPED
[INFO] Keycloak Authz: Parent ............................. SKIPPED
[INFO] Keycloak AuthZ: Policy Provider Parent ............. SKIPPED
[INFO] Keycloak AuthZ: Common Policy Providers ............ SKIPPED
[INFO] Keycloak Dependencies Server All ................... SKIPPED
[INFO] Keycloak Quarkus Parent ............................ SKIPPED
[INFO] Keycloak Configuration API ......................... SKIPPED
[INFO] Keycloak Administration UI ......................... SKIPPED
[INFO] Admin UI REST extensions ........................... SKIPPED
[INFO] Keycloak Quarkus Server Extension .................. SKIPPED
[INFO] Keycloak Quarkus Server Deployment ................. SKIPPED
[INFO] Keycloak Docs Parent ............................... SKIPPED
[INFO] Keycloak Guides Maven Plugin ....................... SKIPPED
[INFO] Keycloak Guides .................................... SKIPPED
[INFO] Keycloak Federation ................................ SKIPPED
[INFO] Keycloak Util Embedded LDAP ........................ SKIPPED
[INFO] Keycloak Util Parent ............................... SKIPPED
[INFO] Keycloak Integration ............................... SKIPPED
[INFO] Keycloak Admin REST Client JavaEE .................. SKIPPED
[INFO] Keycloak Admin REST Client ......................... SKIPPED
[INFO] Keycloak Client Registration API ................... SKIPPED
[INFO] Keycloak Client CLI ................................ SKIPPED
[INFO] Keycloak Admin CLI ................................. SKIPPED
[INFO] Keycloak Client CLI Distribution ................... SKIPPED
[INFO] Keycloak Adapter SPI ............................... SKIPPED
[INFO] Keycloak Undertow Integration SPI .................. SKIPPED
[INFO] Common JBoss/Wildfly Core Classes .................. SKIPPED
[INFO] Keycloak Client Adapter SPI Modules ................ SKIPPED
[INFO] Keycloak SAML Client Adapter Public API ............ SKIPPED
[INFO] Keycloak SAML Client Adapter Core .................. SKIPPED
[INFO] Keycloak SAML Client Adapter Core Jakarta .......... SKIPPED
[INFO] Keycloak Undertow SAML Adapter ..................... SKIPPED
[INFO] Keycloak WildFly Elytron SAML Adapter .............. SKIPPED
[INFO] Keycloak Wildfly SAML Adapter Subsystem ............ SKIPPED
[INFO] Keycloak WildFly Elytron Jakarta SAML Adapter ...... SKIPPED
[INFO] Keycloak Wildfly Jakarta SAML Adapter Subsystem .... SKIPPED
[INFO] Keycloak SAML Wildfly Integration .................. SKIPPED
[INFO] Keycloak SAML Client Adapter Modules ............... SKIPPED
[INFO] Keycloak OIDC Client Adapter Modules ............... SKIPPED
[INFO] Keycloak Adapters .................................. SKIPPED
[INFO] Keycloak Authz: Client API ......................... SKIPPED
[INFO] Keycloak Authz: Policy Enforcer .................... SKIPPED
[INFO] Keycloak Misc ...................................... SKIPPED
[INFO] keycloak-test-helper ............................... SKIPPED
[INFO] Keycloak JavaScript Admin Client ................... SKIPPED
[INFO] Keycloak UI shared components ...................... SKIPPED
[INFO] Keycloak JavaScript Adapter ........................ SKIPPED
[INFO] Keycloak Quarkus Server App ........................ SKIPPED
[INFO] Keycloak Quarkus Server Distribution ............... SKIPPED
[INFO] Keycloak Quarkus Test Parent ....................... SKIPPED
[INFO] Keycloak Quarkus Server JUnit 5 Internal Test Framework SKIPPED
[INFO] Keycloak Quarkus Server Integration tests .......... SKIPPED
[INFO] Keycloak TestSuite ................................. SKIPPED
[INFO] DB Allocator Plugin ................................ SKIPPED
[INFO] Keycloak Arquillian Integration TestSuite .......... SKIPPED
[INFO] Test apps .......................................... SKIPPED
[INFO] Test apps distribution ............................. SKIPPED
[INFO] Keycloak Authz Tests: Hello World Example .......... SKIPPED
[INFO] Keycloak Authz: Servlet Authorization Test ......... SKIPPED
[INFO] Keycloak Authz: Simple Servlet App with Policy Enforcer SKIPPED
[INFO] Servers ............................................ SKIPPED
[INFO] Auth Server ........................................ SKIPPED
[INFO] Auth Server Services ............................... SKIPPED
[INFO] Auth Server Services - Testsuite Providers Deployment SKIPPED
[INFO] Auth Server Services - Testsuite Providers ......... SKIPPED
[INFO] Test Util .......................................... SKIPPED
[INFO] integration-arquillian-test-apps-servlets .......... SKIPPED
[INFO] integration-arquillian-test-apps-servlets-jakarta .. SKIPPED
[INFO] integration-arquillian-test-apps-cors-parent ....... SKIPPED
[INFO] Angular Product Portal JS .......................... SKIPPED
[INFO] JAX-RS Database Service Using OAuth Bearer Tokens .. SKIPPED
[INFO] Testsuite Adapter SPI .............................. SKIPPED
[INFO] Adapter Undertow SPI (JakartaEE) ................... SKIPPED
[INFO] Undertow SAML Adapter (JakartaEE) .................. SKIPPED
[INFO] Keycloak TestSuite Utils ........................... SKIPPED
[INFO] Auth Server - Undertow ............................. SKIPPED
[INFO] Auth Server - Quarkus .............................. SKIPPED
[INFO] App Server ......................................... SKIPPED
[INFO] App Server - SPI ................................... SKIPPED
[INFO] Keycloak Arquillian Integration TestSuite - Cache Server SKIPPED
[INFO] Keycloak Arquillian Integration TestSuite - Cache Server - Infinispan SKIPPED
[INFO] Keycloak Arquillian Integration TestSuite - Cache Server - Legacy SKIPPED
[INFO] Tests .............................................. SKIPPED
[INFO] Base TestSuite ..................................... SKIPPED
[INFO] Other Tests Modules ................................ SKIPPED
[INFO] Tests for logical storage layer .................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  27.687 s
[INFO] Finished at: 2025-01-16T14:19:37-06:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M7:test (default-test) on project keycloak-services:
[ERROR]
[ERROR] Please refer to /home/eric/Documents/github/keycloak/services/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :keycloak-services
```

(no dump files at `keycloak/services/target/surefire-reports`)

</details>


<details>
<summary><code>./mvnw -pl quarkus/deployment,quarkus/dist -am -DskipTests clean install</code> output :x:</summary>

```sh
$ git checkout 25.0.6
$ ./mvnw -pl quarkus/deployment,quarkus/dist -am -DskipTests clean install
...
[INFO] Reactor Summary for Keycloak 25.0.6:
[INFO]
[INFO] Keycloak ........................................... SUCCESS [  0.720 s]
[INFO] Keycloak Common .................................... SUCCESS [  2.367 s]
[INFO] Keycloak Core ...................................... SUCCESS [  1.808 s]
[INFO] Keycloak Server SPI ................................ SUCCESS [  0.925 s]
[INFO] Keycloak Server Private SPI ........................ SUCCESS [  1.447 s]
[INFO] Keycloak Crypto Parent ............................. SUCCESS [  0.015 s]
[INFO] Keycloak Crypto Default ............................ SUCCESS [  0.330 s]
[INFO] Keycloak SAML Core Public API ...................... SUCCESS [  0.369 s]
[INFO] Keycloak SAML Core ................................. SUCCESS [  0.851 s]
[INFO] Keycloak Model Parent .............................. SUCCESS [  0.019 s]
[INFO] Keycloak Database Support .......................... SUCCESS [  0.172 s]
[INFO] Keycloak Database Support - Private classes ........ SUCCESS [  0.462 s]
[INFO] Keycloak REST Services ............................. SUCCESS [  3.822 s]
[INFO] Keycloak Crypto FIPS 140-2 Integration ............. SUCCESS [  0.197 s]
[INFO] Keycloak JavaScript Adapter (JAR) .................. SUCCESS [  5.987 s]
[INFO] Keycloak JavaScript Parent ......................... SUCCESS [  0.043 s]
[INFO] Keycloak Admin UI .................................. SUCCESS [ 20.740 s]
[INFO] Keycloak Account UI ................................ SUCCESS [ 17.174 s]
[INFO] Keycloak Themes .................................... SUCCESS [  5.002 s]
[INFO] Keycloak Model JPA ................................. SUCCESS [  0.856 s]
[INFO] Keycloak Kerberos Federation ....................... SUCCESS [  0.075 s]
[INFO] Keycloak LDAP UserStoreProvider .................... SUCCESS [  0.468 s]
[INFO] Keycloak Database Support - REST services .......... SUCCESS [  0.127 s]
[INFO] Keycloak Model Infinispan .......................... SUCCESS [  1.204 s]
[INFO] Keycloak SSSD Federation ........................... SUCCESS [  0.544 s]
[INFO] Keycloak Authz: Parent ............................. SUCCESS [  0.007 s]
[INFO] Keycloak AuthZ: Policy Provider Parent ............. SUCCESS [  0.007 s]
[INFO] Keycloak AuthZ: Common Policy Providers ............ SUCCESS [  0.145 s]
[INFO] Keycloak Quarkus Parent ............................ SUCCESS [  0.018 s]
[INFO] Keycloak Configuration API ......................... SUCCESS [  0.102 s]
[INFO] Keycloak Administration UI ......................... SUCCESS [  0.005 s]
[INFO] Admin UI REST extensions ........................... SUCCESS [  1.523 s]
[INFO] Keycloak Quarkus Server Extension .................. SUCCESS [  2.371 s]
[INFO] Keycloak Quarkus Server Deployment ................. SUCCESS [  0.752 s]
[INFO] Keycloak Integration ............................... SUCCESS [  0.008 s]
[INFO] Keycloak Client CLI ................................ SUCCESS [  0.006 s]
[INFO] Keycloak Admin CLI ................................. SUCCESS [  1.048 s]
[INFO] Keycloak Client CLI Distribution ................... SUCCESS [  0.405 s]
[INFO] Keycloak Quarkus Server App ........................ FAILURE [  1.924 s]
[INFO] Keycloak Quarkus Server Distribution ............... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:14 min
[INFO] Finished at: 2025-01-16T14:22:01-06:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:3.8.5:build (default) on project keycloak-quarkus-server-app: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step org.keycloak.quarkus.deployment.KeycloakProcessor#configureKeycloakSessionFactory threw an exception: java.util.ServiceConfigurationError: org.keycloak.authentication.AuthenticatorFactory: Provider org.keycloak.authentication.authenticators.browser.UsernamePasswordFormFactory could not be instantiated
[ERROR]         at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:582)
[ERROR]         at java.base/java.util.ServiceLoader$ProviderImpl.newInstance(ServiceLoader.java:809)
[ERROR]         at java.base/java.util.ServiceLoader$ProviderImpl.get(ServiceLoader.java:725)
[ERROR]         at java.base/java.util.ServiceLoader$3.next(ServiceLoader.java:1397)
[ERROR]         at org.keycloak.provider.DefaultProviderLoader.load(DefaultProviderLoader.java:60)
[ERROR]         at org.keycloak.provider.ProviderManager.load(ProviderManager.java:93)
[ERROR]         at org.keycloak.quarkus.deployment.KeycloakProcessor.loadFactories(KeycloakProcessor.java:699)
[ERROR]         at org.keycloak.quarkus.deployment.KeycloakProcessor.configureKeycloakSessionFactory(KeycloakProcessor.java:411)
[ERROR]         at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:580)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
[ERROR]         at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
[ERROR]         at java.base/java.lang.Thread.run(Thread.java:1575)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] Caused by: java.lang.ExceptionInInitializerError
[ERROR]         at org.keycloak.authentication.authenticators.browser.UsernamePasswordForm.<clinit>(UsernamePasswordForm.java:39)
[ERROR]         at org.keycloak.authentication.authenticators.browser.UsernamePasswordFormFactory.<clinit>(UsernamePasswordFormFactory.java:38)
[ERROR]         at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized0(Native Method)
[ERROR]         at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized(Unsafe.java:1161)
[ERROR]         at java.base/jdk.internal.reflect.MethodHandleAccessorFactory.ensureClassInitialized(MethodHandleAccessorFactory.java:340)
[ERROR]         at java.base/jdk.internal.reflect.MethodHandleAccessorFactory.newConstructorAccessor(MethodHandleAccessorFactory.java:103)
[ERROR]         at java.base/jdk.internal.reflect.ReflectionFactory.newConstructorAccessor(ReflectionFactory.java:173)
[ERROR]         at java.base/java.lang.reflect.Constructor.acquireConstructorAccessor(Constructor.java:548)
[ERROR]         at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:498)
[ERROR]         at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:485)
[ERROR]         at java.base/java.util.ServiceLoader$ProviderImpl.newInstance(ServiceLoader.java:785)
[ERROR]         ... 15 more
[ERROR] Caused by: java.lang.IllegalArgumentException: Invalid logger interface org.keycloak.services.ServicesLogger (implementation not found in QuarkusClassLoader:Augmentation Class Loader: PROD for keycloak@517a8c5e)
[ERROR]         at org.jboss.logging.Logger.doGetMessageLogger(Logger.java:2573)
[ERROR]         at org.jboss.logging.Logger.getMessageLogger(Logger.java:2532)
[ERROR]         at org.jboss.logging.Logger.getMessageLogger(Logger.java:2518)
[ERROR]         at org.keycloak.services.ServicesLogger.<clinit>(ServicesLogger.java:50)
[ERROR]         ... 26 more
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :keycloak-quarkus-server-app
```

</details>


via https://github.com/keycloak/keycloak/issues/24320
<details>
<summary><code>./mvnw -f adapters/oidc/js/pom.xml clean install</code> output ✅</summary>

```sh
$ git checkout 25.0.6
$ ./mvnw -f adapters/oidc/js/pom.xml clean install
...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.908 s
[INFO] Finished at: 2025-01-16T14:22:59-06:00
[INFO] ------------------------------------------------------------------------

# Then I wanted to try running Keycloak.
# But this file doesn't exist after that build probably because the above command just builds one dependency
$ java -jar quarkus/server/target/lib/quarkus-run.jar start-dev
Error: Unable to access jarfile quarkus/server/target/lib/quarkus-run.jar
```

</details>

### Anything else?

Related issues:

 - https://github.com/keycloak/keycloak/issues/9573
 - https://github.com/keycloak/keycloak/discussions/9857
 - https://github.com/keycloak/keycloak/discussions/11997
 - https://github.com/keycloak/keycloak/issues/33511

--- a/docs/building.md
+++ b/docs/building.md
@@ -1,9 +1,15 @@
 ## Building from source
 
-Ensure you have JDK 21 (or newer) and Git installed
+Ensure you have **JDK 17** or **JDK 21** and Git installed
 
     java -version
     git --version
+
+Newer versions of the JDK are not supported. If you have multiple JDK versions
+installed, you can specify which one to use during the build by setting the `JAVA_HOME`
+environment variable (this should be the directory containing `/bin/` or `/jre/`).
+
+    JAVA_HOME=/path/to/jdk-21/ ./mvnw clean install
 
 Instead of using a locally installed Maven, call the Maven wrapper script `mvnw` in the main folder of the project.
 This will use the Maven version which is supported by this project.

